test/sheldon_SUITE.erl

```diff
@@ -237,3 +239,95 @@ suggest_words(_Config) ->
     [#{candidates := [], line_number := 1, word := "this_is_wrong"}]
    } = sheldon:check("The next one is this_is_wrong, no suggestions for that"),
   ok.
+
+-spec markdown_adapter(config()) -> ok.
+markdown_adapter(_Config) ->
+  {ok, MarkownFile} =
+    file:read_file([ code:priv_dir(sheldon)
+                   , "/../test/files/markdown.md"
+                   ]),
+
+  #{ bazinga := _
+   , misspelled_words :=
+     [ #{ line_number := 1, word := "API" }
+     , #{ line_number := 1, word := "#" }
+     , #{ line_number := 3, word := "##" }
+     , #{ line_number := 7, word := "##" }
+     , #{ line_number := 9, word := "comments**" }
+     , #{ line_number := 9, word := "**general" }
+     , #{ line_number := 9, word := "**questions**" }
+     , #{ line_number := 12, word := "open-source" }
+     , #{ line_number := 14, word := "##" }
+     , #{ line_number := 16, word := "####" }
+     , #{ line_number := 18, word := "API" }
+     , #{ line_number := 20, word := "open-source" }
+     , #{ line_number := 22, word := "start_block" }
+     , #{ line_number := 24, word := "end_block" }
+     , #{ line_number := 27, word := "start_block1" }
+     , #{ line_number := 27, word := "block*" }
+     , #{ line_number := 27, word := "*another" }
+     , #{ line_number := 28, word := "start_block" }
```

Comment:
You should either document the strict result ordering in README and ensure that in the code or you should not require a specific order in the tests.